### PR TITLE
Refactor/input evnts

### PR DESCRIPTION
**Pull request by skhamvon**
Move input events interfaces to ods-common-core

### CODE REVIEW

File: packages/common/core/src/form/interfaces/events.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import type { OdsValidityState } from '../validation/ods-validity-state';
+import type { EventEmitter } from '@stencil/core';
+
+interface OdsCommonFieldValueChangeEventDetail {
```
Comment (aesteves60) [inline packages/common/core/src/form/interfaces/events.ts:4]: We need the name of the field too

File: packages/components/src/input/src/components/osds-input/interfaces/events.ts [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
-  value: string | undefined | null;
-  oldValue?: string | undefined | null;
-}
-
-type OdsInputValueChangeEvent = CustomEvent<OdsInputValueChangeEventDetail>;
-
-interface OdsInputEvent {
-  odsInputBlur: EventEmitter<void>;
-  odsInputFocus: EventEmitter<void>;
+interface OdsInputEvent extends OdsCommonFieldEvent {
```
Comment (aesteves60) [inline packages/components/src/input/src/components/osds-input/interfaces/events.ts:4]: Don't rewrite the `odsBlur`, `odsFocus` & `odsValueChange`

**Review by aesteves60 — CHANGES_REQUESTED** [review]
Need to add the event odsReset, odsClear and odsHide

File: packages/common/core/src/form/interfaces/events.ts (hunk 1 above)
Comment (skhamvon) [reply packages/common/core/src/form/interfaces/events.ts:4]: added

**Review by aesteves60 — APPROVED** [review]

**Pull request merged**